batavia/utils.js

```diff
@@ -412,30 +447,58 @@ batavia.operators = {
     },
     MULTIPLY: function(a, b) {
         var result, i;
-        if (a instanceof Array) {
+        
+        // If one of the two objects is a list, move it into the first position.
+        if ((b instanceof Array || typeof b == 'string') && !(a instanceof Array || typeof a == 'string')) {
+            var c = b;
+            b = a;
+            a = c;
+        }
+        
+        if (a instanceof Array || typeof a == 'string') {
+            if (b === null){
+                throw new batavia.builtins.TypeError("can't multiply sequence by non-int of type 'NoneType'");
+            }
             result = new batavia.core.List();
-            if (b instanceof Array) {
-                throw new batavia.builtins.TypeError("can't multiply sequence by non-int of type 'list'");
-            } else {
+            if (typeof b == 'number' || typeof b == 'boolean') {
                 for (i = 0; i < b; i++) {
                     result.extend(a);
                 }
+            } else {
+                var bType = batavia.builtins.type([b]).split("'")[1];
+                throw new batavia.builtins.TypeError("can't multiply sequence by non-int of type '" + bType + "'");
             }
-        } else if (b === null){
-            throw new batavia.builtins.TypeError("can't multiply sequence by non-int of type 'NoneType'");
-        } else if (b instanceof Array) {
-            result = [];
-            for (i = 0; i < a; i++) {
-                result.extend(b);
+        } 
+        else if (a != null && b != null 
+            && ((typeof a.valueOf() == 'number' || typeof a == 'boolean')
+                && (typeof b.valueOf() == 'number' || typeof b == 'boolean'))) {
+            if (a instanceof batavia.core.Float || b instanceof batavia.core.Float) {
+                result = new batavia.core.Float(a*b);
+            } else {
+                result = a * b;
             }
-        }
-        else {
-            result = a * b;
+        } else {
+            var aType = batavia.builtins.type([a]).split("'")[1];
+            var bType = batavia.builtins.type([b]).split("'")[1];
+            throw new batavia.builtins.TypeError("unsupported operand type(s) for *: '" + aType + "' and '" + bType + "'");
         }
         return result;
     },
     DIVIDE: function(a, b) {
-        return Math.floor(a / b);
+        if (a != null && b != null 
+            && ((typeof a.valueOf() == 'number' | typeof a == 'boolean')
+                && (typeof b.valueOf() == 'number' || typeof b == 'boolean'))) {
+            if (a instanceof batavia.core.Float || b instanceof batavia.core.Float) {
+                result = new batavia.core.Float(a*b);
+            } else {
+                result = a * b;
+            }
+        } else {
+            var aType = batavia.builtins.type([a]).split("'")[1];
+            var bType = batavia.builtins.type([b]).split("'")[1];
```

Comment:
This will absolutely work - but it involves a linear scan of the content, creating an array with three elements, then throwing away 2 of them. I suspect it would be faster if you just did a `slice(1,-1)` (since we know the quotes are the first and last elements) - that way there's just a single substring being created, based on known string indicies.